app/src/main/java/org/commcare/dalvik/reminders/MainActivity.kt

```diff
@@ -1,44 +1,64 @@
 package org.commcare.dalvik.reminders
 
+import android.app.Activity
 import android.app.AlarmManager
 import android.content.Intent
-import android.content.pm.PackageManager
 import android.os.Build
 import android.os.Bundle
 import android.provider.Settings
 import android.view.View.GONE
 import android.view.View.VISIBLE
-import android.widget.Toast
+import android.widget.Button
+import androidx.activity.result.contract.ActivityResultContracts
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.app.ActivityCompat.*
 import androidx.core.content.ContextCompat
-import androidx.core.content.PermissionChecker
-import androidx.core.content.PermissionChecker.PERMISSION_DENIED
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.recyclerview.widget.GridLayoutManager
-import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_main.remindersRecyclerView
+import kotlinx.android.synthetic.main.activity_main.statusTv
 import org.commcare.dalvik.reminders.ui.RemindersAdapter
+import org.commcare.dalvik.reminders.utils.PermissionUtil
 import org.commcare.dalvik.reminders.viewmodel.ReminderViewModel
 
 class MainActivity : AppCompatActivity() {
 
     companion object {
         private const val COLUMN_COUNT = 2
-        private const val CC_CASE_READ_PERMISSION = "org.commcare.dalvik.provider.cases.read"
-        private const val CC_PERMISSION_REQUEST_CODE = 1
     }
 
     private val ALARM_PERMISSION_REQUEST_CODE: Int = 1111;
     private lateinit var reminderViewModel: ReminderViewModel
+    private lateinit var requestPermissionBtn: Button
+
+    private val startPermissionActivityForResult =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                result.data?.let { intent ->
+                    intent.getStringExtra("isPermissionGranted")?.let {permissionResult ->
+                        if (permissionResult == PermissionActivity.GRANTED) {
+                            getStarted()
+                        } else {
+                            requestPermissionBtn.visibility = VISIBLE
+                            setStatus(R.string.storage_permission_not_granted)
+                        }
+                    }
+                }
+            }
+        }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        requestPermissionBtn = findViewById(R.id.requestSetting)
+        requestPermissionBtn.setOnClickListener {
+            launchPermissionActivity()
+        }
+
```

Comment:
We should do a default permission request on app startup without needing the user to click a button. The button should only be visible if the permission is not granted after the default check.

Reply:
The behaviour is exactly the same.

As the application starts up it checks for alarm premission i.e **validateAlarmPermission()** and once the alarm permission is granted or not required then read permission is intiated automatically.

The "Request Permission" button is only available when the read permission is not granted by user.